src/Player/Player.cpp

```diff
@@ -37,7 +40,145 @@ namespace Player {
+
+                //Checking if cards are valid
+               /* if(this->hand->cardsValidForExchange(*//* IS VALID FOR EXCHANGE *//*)){
+                    exchange_desired = Terminal::print_select("Player [Color] has cards that are valid for exchange, would you like to proceed now?");
+                    if(exchange_desired){
+                        army_exchange += this->hand->exchange(*//*  INDICES *//* );
+                    }
+                }*/
+
+                //Exchanging cards
+                while(this->hand->size() >= 5){
+                    army_exchange += this->hand->exchange(&answer_cards[0]); /* ??????? check if conversion is correct */
+                }
+            }
+
+        //Therefore, the total armies available (A) to Player are
+        army_total = army_country + army_continent + army_exchange;
+
+        /*
+         * Then player places all armies on some of the countries it owns as they see fit.
+         */
+
+        //Looping around the number of armies available to place
+        do{
+            //Display available options (armies available + countries for positioning)
+            Terminal::print("You are now able to place your reinforcement armies. The total armies and countries available for positioning are the following: ");
+                vector<string> armies_print;
+                armies_print.emplace_back(army_total + " total armies available");
+                Terminal::print(armies_print);
+            /*INDEX = */
+
+                //Gathering countries owned by player and in map
+                    //vector<Countries *> countries_owned ~ already declared above
+                    //vector<Countries *> countries_in_map ~ already declared above
+                //Printing Utility
+                vector<string> options;
+                for(auto c : countries_owned) {
+                    options.emplace_back(c->get_name() + " (which has" + to_string(c->get_armies()) + " armies present).");
+                }
+                //Asking player to input data
+                options.emplace_back("\nPlease enter the country [#] to which you would like to place armies: ");
+                int answer_country = Terminal::print_select(options);
+
+                vector<string> options_number;
+                options_number.emplace_back("Enter how many armies you would like to place on " + countries_in_map.at(placement_answer)->get_name());
+                int answer_number = Terminal::print_select(options_number);
+
+                //Adding 'answer_number' armies to 'answer_country'
+                countries_in_map.at(answer_country)->set_armies(countries_in_map.at(answer_country)->get_armies() + answer_number);
+
+            //Reducing the # of armies available
+            army_total -= answer_number;
+        }while(army_total !=0);
```

Comment:
army_total <= 0; As of right now, if the user enter a huge number, it will go pass 0, and won't ever be equal => infinite loop

Reply:
Thanks. Fixed.